File: 57_insert_interval/insert.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        n = len(intervals)

        for i in range(n):
            if newInterval[1] < intervals[i][0]:
                res.append(newInterval)
                res += intervals[i:]
                return res
            elif newInterval[0] > intervals[i][1]:
                res.append(intervals[i])
            else:
                start = min(newInterval[0], intervals[i][0])
                end = max(newInterval[1], intervals[i][1])
                newInterval = [start, end]

        res.append(newInterval)

        return res

File: 57_insert_interval/insert.go
func insert(intervals [][]int, newInterval []int) [][]int {
    var res [][]int

    for idx, interval := range intervals {
        if interval[0] > newInterval[1] {
            res = append(res, newInterval)
            res = append(res, intervals[idx:]...)
            return res
        } else if interval[1] < newInterval[0] {
            res = append(res, interval)
        } else {
            start := min(interval[0], newInterval[0])
            end := max(interval[1], newInterval[1])
            newInterval = []int {start, end}
        }
    }
    
    res = append(res, newInterval)

    return res
}

func max(a int, b int) int {
    if a > b {
        return a
    }

    return b
}

func min(a int, b int) int {
    if a < b {
        return a
    }

    return b
}